### DISCUSSION TITLE
Is it possible to configure an ignore directory or file pattern for spell checking?

### DISCUSSION THREAD

**Discussion by gerardbosch** [valentjn/vscode-ltex#420] [Q&A] [answered]
Hi there,

I'm using the 'Local History' `xyz.local-history` extension, which keeps local version history of files (no Git or VCS involved). This extension works by writing modified files into `PROJECT_DIR/.vscode/.history/FILE_TIMESTAMP.EXT` in my case.

I've got the `.vscode` in my `.gitignore`, as it is a local history.

I've just installed the `LTeX` extension and when I run `LTeX: Check all documents in workspace` it takes ages, as it's spell checking all versions stored under `.vscode/.history/`.

Is there any way to instruct LTeX to ignore a directory, a file pattern, or to ignore stuff listed in `.gitignore` for example?

Thanks!

**Comment by valentjn** [comment, marked as answer]
I haven't tried it, but adding such directories to VS Code's `files.exclude` should work. That's also the canonical setting for this.

Edit: Just tried it, it works.

**Reply by timburgan** [reply]
@valentjn Can you please help explain the above solution to me? Won't adding a directory (whose files I was excluded from ltex) to `files.exclude` cause that whole directory to be hidden from the Explorer tree and any VS Code searches?

That was this seems to do for me when I try it.

For instance, my workspace is full of markdown files, and I also have a subdirectory `quick_notes` that I don't want ltex to spell check when the files open or while I'm editing, yet I still want these documents available in the Explorer tree, etc.
